cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/queue/RoundRobinQueueReader.java

```diff
@@ -36,12 +36,18 @@
 public final class RoundRobinQueueReader<T> extends TimeTrackingQueueReader<T> {
 
   private final InputDatum<T> nullInput = new NullInputDatum<>();
+  private final Iterable<QueueReader<T>> readerIterable;
   private final Iterator<QueueReader<T>> readers;
 
   public RoundRobinQueueReader(Iterable<QueueReader<T>> readers) {
+    this.readerIterable = readers;
     this.readers = Iterables.cycle(readers).iterator();
   }
 
+  public Iterable<QueueReader<T>> getReaders() {
```

Comment:
No a fan of exposing fields like this. We should be able to do the checking from the caller side (i.e. the `FlowletProgramRunner`) before calling the constructor of this class.